atomic_reactor/inner.py

```diff
@@ -538,7 +536,53 @@ def save(self, context_dir: ContextDir) -> None:
 
         logger.info("Writing workflow data into %s", context_dir.workflow_json)
         with open(context_dir.workflow_json, "w+") as f:
-            json.dump(self.dump(), f)
+            json.dump(asdict(self), f, cls=WorkflowDataEncoder)
```

Comment:
As you mentioned, `dataclasses.asdict()` converts nested data classes recursively. That could cause issues if some of the WorkflowData attributes were dataclasses

Reply:
Restored the ISerializer interface for workflow data object.